### PR TITLE
Serial

### PR DESCRIPTION
**Pull request by jgkamat**
This should have been merged a while ago @rmkeezer

### CODE REVIEW

**Comment by jgkamat** [comment]
This should probably be merged before the demo today.

**Comment by jgkamat** [comment]
@rmkeezer

File: src/main/java/io/github/mountainrange/mule/gameplay/WorldMap.java [hunk 1]
```diff
@@ -5,19 +5,22 @@
...
-public class WorldMap<T extends Tile> implements Iterable<Tile> {
+public class WorldMap<T extends Tile> implements Iterable<Tile>, Serializable {
 
 	/** Backing instance of Grid. */
 	private Grid<T> baseGrid;
 
-	/** List of tiles owned by each player, sorted by the MULE installed on them. */
-	private Map<Player, EnumMap<MuleType, Set<Tile>>> productionTiles;
+	/* List of tiles owned by each player, sorted by the MULE installed on them */
+	private Map<Player, EnumMap<MuleType, Set<Point>>> productionTiles;
```
Comment (strburst) [inline src/main/java/io/github/mountainrange/mule/gameplay/WorldMap.java:23]: Would it be possible to keep this just a `Set<Tile>`? We should probably make this transient, and just rebuild it after we finish loading `baseGrid` by looping through all the tiles once and putting the `Tile`s in the right `Set`s.

File: src/main/java/io/github/mountainrange/mule/gameplay/Grid.java [hunk 2]
```diff
@@ -70,6 +73,42 @@ public T get(int column, int row) {
...
+	public Point get(T t) {
+		for (int i = 0; i < grid.length; i++) {
+			for (int j = 0; j < grid[0].length; j++) {
+				if (grid[i][j].equals(t)) {
+					return new Point(i, j);
+				}
+			}
+		}
+		return new Point(0, 0);
+	}
```
Comment (strburst) [inline src/main/java/io/github/mountainrange/mule/gameplay/Grid.java:98]: This is very inefficient, and could be avoided; see my other comment.

File: src/main/java/io/github/mountainrange/mule/gameplay/WorldMap.java (hunk 1 above)
Comment (rmkeezer) [inline src/main/java/io/github/mountainrange/mule/gameplay/WorldMap.java:23]: At the time I didn't see if there was another way that worked or not, if you can find a better solution that works with the tests please do!

Comment (strburst) [inline src/main/java/io/github/mountainrange/mule/gameplay/WorldMap.java:23]: Don't worry, I've been messing with `Serializable` lately. You can override `readObject` and `writeObject`, but you can also do `in.defaultReadObject` and `out.defaultWriteObject` to take advantage of Java automagically serializing instance variables that aren't transient, and then do some initialization afterwards to initialize transient fields. I've had some JavaFX initialization issues that I've been trying to debug, though.

**Comment by jgkamat** [comment]
Guys let's get this merged please

**Comment by joshhting** [comment]
Is it all set for merging?

**Comment by jgkamat** [comment]
Does the rectangle save feature work?
if so, go ahead and merge asap
EDIT: probably not :/

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Code Review Recommended. Check Details Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=MountainRange&toolchainId=matthew_test_please_delete&branch=serial&env_id=ibm:ys1:us-south

**Comment by jgkamat** [comment]
> Devops Insights Suggestion
> Code Review Recommended. Check Details Click here for more info:
> https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=MountainRange&toolchainId=matthew_test_please_delete&branch=serial&env_id=ibm:ys1:us-south

Tbh the code is probably not worth reviewing anymore lol also I think IBM
hacked you kthxby

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=MountainRange&toolchainId=matthew_test_please_delete&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=8779e7b4-5ccd-432f-b1fe-8596a35f3bc9&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=8779e7b4-5ccd-432f-b1fe-8596a35f3bc9&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=8779e7b4-5ccd-432f-b1fe-8596a35f3bc9&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=4e072043-130a-41da-b0ba-2875f9e67103&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://console.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=ecd9bfdb-842a-4e98-9b99-0c8574f6f076&branch=serial&env_id=ibm:yp:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=matthew_test_please_delete&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=8779e7b4-5ccd-432f-b1fe-8596a35f3bc9&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=matthew_test_please_delete&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://console.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=ecd9bfdb-842a-4e98-9b99-0c8574f6f076&branch=serial&env_id=ibm:yp:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=matthew_test_please_delete&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=matthew_test_please_delete3&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=039d2fa9-21bd-4bde-8d5e-a16dbf33a442&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=a5724662-77c3-4fd1-9b15-83b6950aacd6&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=a5724662-77c3-4fd1-9b15-83b6950aacd6&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=matthew_test_please_delete&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=3dc79982-8427-4452-9306-b33f537110b1&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=3dc79982-8427-4452-9306-b33f537110b1&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=ec6ab45d-93e6-44bd-b8b0-4d927e3d1119&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=3dc79982-8427-4452-9306-b33f537110b1&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=ec6ab45d-93e6-44bd-b8b0-4d927e3d1119&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=33dd0584-928b-4556-8139-2d9b045b1be3&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=3dc79982-8427-4452-9306-b33f537110b1&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=94b68ac8-33f5-4520-b4a1-840a85dcfac4&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=3dc79982-8427-4452-9306-b33f537110b1&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=3dc79982-8427-4452-9306-b33f537110b1&branch=serial&env_id=ibm:ys1:us-south

**Comment by rmkeezer** [comment]
**Devops Insights Suggestion**
Devops Insights detected that highly error prone files were touched by this pull request, code review is recommended. Click here for more info:
https://dev-console.stage1.bluemix.net/devops/insights/statuscheck?pull_id=60.0&repo=MULE&account=rmkeezer&toolchainId=3dc79982-8427-4452-9306-b33f537110b1&branch=serial&env_id=ibm:ys1:us-south

**Pull request closed without merging**